src/main/java/com/productdock/library/rental/scheduled/CancelReservation.java

```diff
@@ -64,10 +55,10 @@ private boolean timeToCancelReservation(Date reservationTime) {
     }
 
     private Date getExecutionTime(Date reservationTime) {
-        var executionTime = reservationTime.getTime() + TimeUnit.SECONDS.toMillis(delay);
+        var executionTime = reservationTime.getTime() + reservationHoldPolicy.getTimeUnit().toMillis(reservationHoldPolicy.getLimit());
         DateRange dateRange = new DateRange(reservationTime.getTime(), executionTime);
-        if (dateRange.includesWeekend()) {
-            executionTime += TimeUnit.SECONDS.toMillis(timeToSkipWeekend);
+        if (dateRange.includesWeekend() && reservationHoldPolicy.isSkippedWeekend()) {
+            executionTime += reservationHoldPolicy.getTimeUnit().toMillis(WEEKEND_DAYS);
```

Comment:
`reservationHoldPolicy.getTimeUnit().toMillis(WEEKEND_DAYS)` 
WEEKEND_DAYS implies that it will always be days, it is not configurable by app yaml.
Use TimeUnit.DAYS.toMIllis(WEEKEND_DAYS) directly.